### ISSUE TITLE
[Autocomplete] InputProps adornments incompatibility with multiple

### ISSUE THREAD

**Issue by jgabriele** [lookfirst/mui-rff#469]
**Describe the bug**
We cannot use adornments with `mui-rff`'s Autocomplete

```typescript
<Autocomplete
  multiple
  name="name"
  label="Label"
  options={['A', 'B', 'C', 'D']}
  textFieldProps={{
    variant: 'outlined',
    // This works, but then the autocomplete chips are not shown when an option is selected
    InputProps: {
      startAdornment: (
        <InputAdornment position="start">START</InputAdornment>
      ),
    },
  }}
  className={classes.fieldGroup}
/>
```

This issue has already been [open on MUI side](https://github.com/mui-org/material-ui/issues/22103) and they explain it breaks because the Autocomplete uses `startAdornment` to show the chips. The same issue appears with `endAdornment` as it breaks the ❌ clear button.

The fix they propose is the following:

```diff
<TextField
  {...params}
  label="Proof of Concept"
  fullWidth
  InputProps={{
    ...params.InputProps,
-   startAdornment: "test"
+   startAdornment: <>test{params.InputProps.startAdornment}</>
 }}
/>
```

**To Reproduce**
Render an autocomplete as described above.

**Expected behavior**
- Autocomplete works with `startAdornment` (the user defined one is shown first, then the chips)
- Autocomplete works with `endAdornment` (mui-rff one(s) are shown first, then the user defined one)

**Desktop (please complete the following information):**
 - OS: MacOS 11.4
 - Browser Chrome
 - Version Latest
 
 **Additional context**
Note: I tried to run the project locally as the fix seems pretty straightforward but the [Contributing](https://github.com/lookfirst/mui-rff/blob/master/CONTRIBUTING.md) page and the [Building section](thehttps://github.com/lookfirst/mui-rff#building) did not work. `yarn start` does not spin a server. You might want to update it if needed.

**Comment by lookfirst** [comment]
I just https://github.com/lookfirst/mui-rff/pull/466 released something that mucks with the InputProps handling, can you please report the version of mui you tried?

With regards to the contributing page, it doesn't have anything on it. There is the [building section](https://github.com/lookfirst/mui-rff#building) of the readme and the instructions there work for me. Can you tell me what the error is? `yarn start` needs to be run in the `examples/` folder to spin up a server.

**Comment by jgabriele** [comment]
I was actually running the version `3.0.6` of `mui-rff` and I could not see my adornments at all. Then I upgraded to `3.0.7` (didn't know the timing was so perfect 😄).

Now I see them but as I explained above, `startAdornment` is competing with the chips of the `multiple` autocomplete and `endAdornment` with the X to clear the field.

See [this sandbox](https://codesandbox.io/s/snowy-glitter-lpv81?file=/src/App.js) to see how it looks like.

And yeah, I'm stupid, I was running `yarn start` at the root, not in `examples/` 🤦‍♂️. I'll have a look.
